game-headed/src/main/java/org/triplea/game/client/ui/javafx/util/JavaFxSelectionComponentFactory.java

```diff
@@ -368,27 +373,33 @@ public String validValueDescription() {
     }
 
     @Override
-    public Map<GameSetting<?>, String> readValues() {
-      final Map<GameSetting<?>, String> values = new HashMap<>();
+    public Map<GameSetting<?>, Object> readValues() {
+      final Map<GameSetting<?>, Object> values = new HashMap<>();
+
       if (noneButton.isSelected()) {
-        values.put(proxyChoiceClientSetting, HttpProxy.ProxyChoice.NONE.toString());
+        values.put(proxyChoiceClientSetting, HttpProxy.ProxyChoice.NONE);
       } else if (systemButton.isSelected()) {
-        values.put(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_SYSTEM_SETTINGS.toString());
+        values.put(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_SYSTEM_SETTINGS);
         HttpProxy.updateSystemProxy();
       } else {
-        values.put(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_USER_PREFERENCES.toString());
-        values.put(proxyHostClientSetting, hostText.getText().trim());
-        values.put(proxyPortClientSetting, portText.getText().trim());
+        values.put(proxyChoiceClientSetting, HttpProxy.ProxyChoice.USE_USER_PREFERENCES);
       }
+
+      final String host = hostText.getText().trim();
+      values.put(proxyHostClientSetting, host.isEmpty() ? null : host);
+
+      final String encodedPort = portText.getText().trim();
+      values.put(proxyPortClientSetting, encodedPort.isEmpty() ? null : Integer.valueOf(encodedPort));
```

Comment:
Extracting these `put()` calls outside the preceding `else` block is a bug fix, unrelated to this PR.